packages/devtools_app/lib/src/utils.dart

```diff
@@ -693,6 +694,71 @@ class DebugTimingLogger {
   }
 }
 
+/// Compute a simple moving average.
+class MovingAverage {
+  MovingAverage(
+      {this.averagePeriod = 50, this.ratioSpike = 2, List<int> newDataSet}) {
+    if (newDataSet != null) {
+      var initialDataSet = newDataSet;
+      final count = newDataSet.length;
+      if (count > averagePeriod) {
+        initialDataSet = newDataSet.sublist(count - averagePeriod);
+      }
+
+      dataSet.addAll(initialDataSet);
+      for (final value in dataSet) {
+        averageSum += value;
+      }
+    }
+  }
+
+  final dataSet = Queue<int>();
+
+  /// Total collected items in the X axis (time) used to compute moving average.
+  /// Default 100 periods for memory profiler 1-2 periods / seconds.
+  final int averagePeriod;
+
+  /// Ratio of first item in dataSet when comparing to last - mean
+  /// e.g., 2 is 50% (dataSet.first ~/ ratioSpike).
+  final int ratioSpike;
```

Comment:
nit: should ratioSpike be a double? Callers could then round if need be.

Reply:
Made it a double.  I'm only using as a int.